src/realm/deppart/byfield_gpu_impl.hpp

```diff
@@ -0,0 +1,268 @@
+
+    //Reset counters
+    CUDA_CHECK(cudaMemsetAsync(d_inst_counters, 0, (field_data.size()) * sizeof(uint32_t), stream), stream);
+
+    //Second pass: recompute intersection, but this time write to output
+    grid_size = (parent_entries.size() * inst_size + flattened_threads - 1) / flattened_threads;
+    intersect_input_rects<N,T><<<grid_size, flattened_threads, 0, stream>>>(d_inst_entries, d_parent_entries, d_inst_offsets, d_inst_prefix, inst_size, parent_entries.size(), field_data.size(), d_inst_counters, d_valid_rects);
+    KERNEL_CHECK(stream);
+
+
+    // Prefix sum the valid rectangles by volume
+    size_t* d_prefix_rects = reinterpret_cast<size_t*>(AffineAccessor<char,1>(prefix_rects_instance, 0).base);
+    CUDA_CHECK(cudaMemsetAsync(d_prefix_rects, 0, sizeof(size_t), stream), stream);
+
+    // Build the CUB transform‐iterator
+    using VolIter = cub::TransformInputIterator<
+                      size_t,              // output type
+                      RectFieldVolumeOp<N,T>,            // functor
+                      Rect<N,T>*     // underlying input iterator
+                    >;
+    VolIter d_volumes(d_valid_rects, RectFieldVolumeOp<N,T>());
+
+    void*   d_temp = nullptr;
+    size_t rect_temp_bytes = 0;
+    cub::DeviceScan::InclusiveSum(
+        /* d_temp_storage */  nullptr,
+        /* temp_bytes */      rect_temp_bytes,
+        /* d_in */            d_volumes,
+        /* d_out */           d_prefix_rects + 1,   // shift by one so prefix[1]..prefix[n]
+        /* num_items */       num_valid_rects, stream);
+
+    RegionInstance temp_instance = this->realm_malloc(rect_temp_bytes, my_mem);
+    d_temp = reinterpret_cast<void*>(AffineAccessor<char,1>(temp_instance, 0).base);
+    cub::DeviceScan::InclusiveSum(
+        /* d_temp_storage */  d_temp,
+        /* temp_bytes */      rect_temp_bytes,
+        /* d_in */            d_volumes,
+        /* d_out */           d_prefix_rects + 1,
+        /* num_items */       num_valid_rects, stream);
+
+
+    //Number of points across all rectangles (also our total output count)
+    size_t total_pts;
+    CUDA_CHECK(cudaMemcpyAsync(&total_pts, &d_prefix_rects[num_valid_rects], sizeof(size_t), cudaMemcpyDeviceToHost, stream), stream);
+
+    CUDA_CHECK(cudaStreamSynchronize(stream), stream);
+    inst_entries_instance.destroy();
+    parent_entries_instance.destroy();
+    inst_offsets_instance.destroy();
+    temp_instance.destroy();
+    inst_counters_instance.destroy();
+
+    RegionInstance points_instance = this->realm_malloc(total_pts * sizeof(PointDesc<N,T>), my_mem);
+    PointDesc<N,T>* d_points = reinterpret_cast<PointDesc<N,T>*>(AffineAccessor<char,1>(points_instance, 0).base);
+
+    FT* d_colors;
+    RegionInstance colors_instance;
+
+    //Memcpying a boolean vector breaks things for some reason so we have this disgusting workaround
+    if constexpr(std::is_same_v<FT,bool>) {
```

Comment:
Is making `FT` a bool even allowed?

Reply:
The reason for this though is that `vector<bool>` is specialized to an implementation that packs 8 elements into a single byte, rather than a byte per element.